### PR TITLE
view distance

### PR DESCRIPTION
**Pull request by kashike**
[**SpongeAPI**](https://github.com/SpongePowered/SpongeAPI/pull/1659) | [SpongeCommon](https://github.com/SpongePowered/SpongeCommon/pull/1557)
adds #1658

### CODE REVIEW

File: src/main/java/org/spongepowered/api/world/World.java [hunk 1]
```diff
@@ -447,4 +446,20 @@ default void setSerializationBehavior(SerializationBehavior behavior) {
      * @throws IOException If the save failed
      */
     boolean save() throws IOException;
+
+    /**
+     * Gets the view distance for this world.
```
Comment (Deamon5550) [inline src/main/java/org/spongepowered/api/world/World.java]: We should specify that the value is in chunks

**Review by Deamon5550 — CHANGES_REQUESTED** [review]

File: src/main/java/org/spongepowered/api/world/World.java [hunk 2]
```diff
@@ -447,4 +446,20 @@ default void setSerializationBehavior(SerializationBehavior behavior) {
...
+    int getViewDistance();
+
+    /**
+     * Sets the view distance for this world.
+     *
+     * <p>The view distance must be greater than or equal to 3, andless than or equal to 32.</p>
+     *
+     * @param viewDistance The view distance
+     */
+    void setViewDistance(final int viewDistance);
```
Comment (Deamon5550) [inline src/main/java/org/spongepowered/api/world/World.java:464]: Also we have no way of unsetting a world's view distance (setting it to use the global setting rather).

Comment (ryantheleach) [reply src/main/java/org/spongepowered/api/world/World.java:464]: as per https://github.com/SpongePowered/SpongeCommon/pull/1557#discussion-diff-142562964R107 sending a value of -1 resets it. 

**if** this remains in the API, it needs to be documented.

Comment (Deamon5550) [reply src/main/java/org/spongepowered/api/world/World.java:464]: we can probably do better than that

**Comment by Zidane** [comment]
@kashike 

This needs to be updated to address @Deamon5550 's concerns above.

**Pull request merged**